### DISCUSSION TITLE
Aggregate function with SAMPLE BY for multiple symbols.

### DISCUSSION THREAD

**Discussion by peter-facko** [questdb/questdb#4955] [Ideas]
Hello,

I have a table with price changes of multiple symbols.
Therefore the columns are `TIMESTAMP`, `SYMBOL`, and `VALUE`.
I need OHLC.

If I only had `TIMESTAMP` and `VALUE`, I would use `first(VALUE)` and `SAMPLE BY` for open prices.
But when I also have the `SYMBOL` column, how do I get the open price _for each symbol_?

**Comment by bluestreak01** [comment]
it might be 

```sql
select timestamp, symbol, first(value), last(value),min(value),max(value() from tab sample by 5m
```

**Comment by peter-facko** [comment]
Oh, sorry, I'm dumb. I thought that `SAMPLE BY` is already doing the aggregation but I guess that the implicit `GROUP BY` is at work here.
The code you provided works, thank you. I have no idea why I didn't try it.
